packages/docs-theme/src/components/baseExample.tsx

```diff
@@ -16,11 +17,50 @@ export interface IBaseExampleProps {
  * Starter class for all React example components.
  * Examples and options are rendered into separate containers.
  */
-export class BaseExample<S> extends React.PureComponent<IBaseExampleProps, S> {
+export class BaseExample<S> extends React.Component<IBaseExampleProps, S> {
     /** Define this prop to add a className to the example container */
     protected className: string;
 
+    // Can't put this in state, because the state typing is generic.
+    private hasDelayedBeforeInitialRender = false;
+    private hasCompletedInitialRender = false;
+
+    public shouldComponentUpdate(nextProps: IBaseExampleProps, nextState: S & object) {
+        return (
+            // HACKHACK: Permit one redundant re-render after the inital delay. (This will be the first proper render of
+            // the component.)
+            (this.hasDelayedBeforeInitialRender && !this.hasCompletedInitialRender) ||
+            // Now, mimic PureComponent shouldComponentUpdate behavior:
+            !Utils.shallowCompareKeys(this.props, nextProps) ||
+            !Utils.shallowCompareKeys(this.state, nextState)
+        );
+    }
+
+    public componentWillMount() {
+        // HACKHACK: The docs app suffers from a Flash of Unstyled Content that causes some 'width: 100%' examples to
+        // render incorrectly, because they mis-measure the horizontal space available to them. Until that bug is squashed,
+        // this is the workaround: delay initial render with a requestAnimationFrame.
+        requestAnimationFrame(() => {
+            this.hasDelayedBeforeInitialRender = true;
+            this.forceUpdate();
```

Comment:
hmm i'm not sure this belongs in `WillMount`, feels unsafe to kick off an update before it has mounted. can we move this logic to `DidMount`?

Reply:
I tried that, and it didn't work, presumably because child components may be returning `shouldComponentUpdate => false` for themselves and not changing anything. I really had to ensure that the initial render of the child components was a healthy one.